### PR TITLE
autoformat & pre-commit docs

### PR DESCRIPTION
**Pull request by snopoke**
Cross: https://github.com/czue/pegasus/pull/59

### CODE REVIEW

File: code-structure.md [hunk 1]
```diff
@@ -50,11 +50,11 @@ The module starts with up to five apps, depending on your configuration.
 
 This is where the Pegasus examples live.
 
-In general it is not expected that you'll need to modify much in this module, though feel free to do so!
+In general, it is not expected that you'll need to modify much in this module, though feel free to do so!
 
 ## The `requirements` folder
 
-This is where you define your project's Python requrements.
+This is where you define your project's Python requirements.
```
Comment (czue) [inline code-structure.md:57]: :man_facepalming:

File: code-structure.md [hunk 2]
```diff
@@ -77,3 +77,25 @@ from the `assets` folder as well as images.
...
 The majority of the project's base template layouts are in the `templates/web` folder.
+
+## Code formatting
+
+For projects that have enabled the `Autoformat code` option, the code will have been formatted
+using [black](https://black.readthedocs.io/en/stable/) and [isort](https://pycqa.github.io/isort/).
+
+The project will also include [pre-commit](https://pre-commit.com/) as a dependency in the requirements file
+as well as the `.pre-commit-config.yaml` file in the root directory. pre-commit is a tool for managing pre-commit
+hooks.
```
Comment (czue) [inline code-structure.md:88]:
```suggestion
hooks - which can be used to ensure your code matches the correct format when it's committed.
```

(not 100% sure if this is right, feel free to tweak).

File: release-notes.md [hunk 3]
```diff
@@ -3,7 +3,7 @@ Version History and Release Notes
 
 Releases of [SaaS Pegasus: The Django SaaS Boilerplate](https://www.saaspegasus.com/) are documented here.
 
-## Version 2023.1
+## Version 2023.2
```
Comment (czue) [inline release-notes.md:6]: i thought i already fixed this (https://docs.saaspegasus.com/release-notes) :thinking:

**Review by czue — COMMENTED** [review]
This should probably be a PR into `develop` yeah?

**Review by czue — APPROVED** [review]

File: code-structure.md [hunk 4]
```diff
@@ -77,3 +77,64 @@ from the `assets` folder as well as images.
...
+
+The default configuration that ships with Pegasus will run `isort` and `black` prior to every Git
+commit. If there are fixes that are needed you will be notified in the shell output.
+
+### pre-commit Usage
+
+**Manually running hooks**
+```shell
+# run all hooks against currently staged files
+pre--commit run
```
Comment (czue) [inline code-structure.md:105]:
```suggestion
pre-commit run
```

File: code-structure.md [hunk 5]
```diff
@@ -77,3 +77,64 @@ from the `assets` folder as well as images.
...
+# run all the hooks against all the files. This is a useful invocation if you are using pre-commit in CI.
+pre-commit run --all-files
+# run the isort hook against all staged files
```
Comment (czue) [inline code-structure.md:108]: is it *just* the `isort`? or isort doesn't run unless you run this?

File: code-structure.md [hunk 6]
```diff
@@ -77,3 +77,64 @@ from the `assets` folder as well as images.
...
+
+**Temporarily disable hooks**
+See https://pre-commit.com/#temporarily-disabling-hooks
+
+For more information on using and configuring pre-commit check out the 
+[pre-commit docs](https://pre-commit.com/#quick-start) 
+
+
+### Tool configurations
+The configuration for the tools can be found in the [`pyproject.toml`][toml] file.
```
Comment (czue) [inline code-structure.md:120]: curious about this comment vs this https://github.com/czue/pegasus/pull/59/files#r1097220012

**Review by czue — COMMENTED** [review]
looks great, just trying to wrap my head around some potential inconsistencies

File: code-structure.md (hunk 5 above)
Comment (snopoke) [reply code-structure.md:108]: It is *just* isort. I'll update the wording.

Comment (snopoke) [reply code-structure.md:108]:
```suggestion
# only run the isort hook against all staged files
```

File: code-structure.md (hunk 6 above)
Comment (czue) [reply code-structure.md:120]: This seems fine for now

**Review by czue — APPROVED** [review]

**Pull request merged**